### PR TITLE
Rename dataset

### PR DESCRIPTION
**Pull request by NakulManchanda**
Added capability to add rename to dataset

### CODE REVIEW

File: WebContent/js/utilities/fileHelper.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    }
+    return { DSName: file.substring(0, end),
+        DSMember: file.substring(file.indexOf('(') + 1, file.indexOf(')')) };
+}
+
+export function hasMember(file) {
+    return file !== undefined && file.indexOf('(') > 1;
+}
+
+export function updateDSName(file, newDSName) {
```
Comment (jordanCain) [inline WebContent/js/utilities/fileHelper.js]: As far as I can tell this function is only used by the editor reducer therefore it should stay in that file for simplicity sake. As it's only string manipulation and used in one place it probably doesn't even need to be a function.

File: WebContent/js/actions/treeDatasets.js [hunk 2]
```diff
@@ -222,6 +248,21 @@ function cleanupStateAfterDelete(DSName, isOpenInViewer) {
...
 }
 
+function cleanupStateAfterRename(oldName, newName, isOpenInViewer) {
+    return dispatch => {
+        // Now refresh the datasets members
+        if (isDatasetMember(oldName)) {
+            dispatch(fetchDSMembers(oldName.substring(0, oldName.indexOf('('))));
+        }
+        dispatch(renameDatasetRefresh(oldName, newName));
+        // If we're deleting something that's open in the content viewer we need to close it
```
[the remark this answers is not in the document]
Comment (NakulManchanda) [reply WebContent/js/actions/treeDatasets.js]: good catch, refactored

File: WebContent/js/actions/treeDatasets.js [hunk 3]
```diff
@@ -222,6 +248,21 @@ function cleanupStateAfterDelete(DSName, isOpenInViewer) {
     };
 }
 
+function cleanupStateAfterRename(oldName, newName, isOpenInViewer) {
+    return dispatch => {
+        // Now refresh the datasets members
+        if (isDatasetMember(oldName)) {
```
[the remark this answers is not in the document]
Comment (NakulManchanda) [reply WebContent/js/actions/treeDatasets.js]: moved out function as suggested, thanks!

File: WebContent/js/utilities/urlUtils.js [hunk 4]
```diff
@@ -48,6 +48,18 @@ export function atlasDelete(endpoint) {
     return atlasAction(endpoint, fetchParams);
 }
 
+export function atlasRename(endpoint, newName) {
```
[the remark this answers is not in the document]
Comment (NakulManchanda) [reply WebContent/js/utilities/urlUtils.js]: modified atlasPut, now body should be provided by called as argument, refactored

File: WebContent/js/components/dialogs/datasets/RenameDatasetDialog.js [hunk 5]
```diff
@@ -0,0 +1,31 @@
+/**
```
[the remark this answers is not in the document]
Comment (NakulManchanda) [reply WebContent/js/components/dialogs/datasets/RenameDatasetDialog.js]: good catch, deleted excessive dialog, title now function of member to be renamed

File: WebContent/js/utilities/fileHelper.js (hunk 1 above)
Comment (NakulManchanda) [reply WebContent/js/utilities/fileHelper.js]: i guess for now we can do that, moved in editor.js reducer

File: WebContent/js/components/dialogs/datasets/RenameDialog.js [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+        });
+    }
+
+    render() {
+        const { title, dialogReturn, dispatch, oldName, isOpenInViewer } = this.props;
+
+        const dialogContent = (<UpperCaseTextField
+            placeholder={'New name'}
+            fieldChangedCallback={this.updateName}
+            value={this.state.newName}
```
Comment (jordanCain) [inline WebContent/js/components/dialogs/datasets/RenameDialog.js]: make this dialog fullWidth={true}

File: WebContent/js/components/dialogs/datasets/RenameDatasetDialog.js [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+ * This program and the accompanying materials are made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Copyright IBM Corporation 2016, 2020
```
Comment (jordanCain) [inline WebContent/js/components/dialogs/datasets/RenameDatasetDialog.js]: This file didn't exist in 2016

File: WebContent/js/components/dialogs/datasets/RenameDialog.js [hunk 8]
```diff
@@ -0,0 +1,71 @@
+/**
+ * This program and the accompanying materials are made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Copyright IBM Corporation 2016, 2020
```
Comment (jordanCain) [inline WebContent/js/components/dialogs/datasets/RenameDialog.js]: File didn't exist in 2016

File: WebContent/js/utilities/fileHelper.js [hunk 9]
```diff
@@ -0,0 +1,27 @@
+/**
+ * This program and the accompanying materials are made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Copyright IBM Corporation 2016, 2020
```
Comment (jordanCain) [inline WebContent/js/utilities/fileHelper.js]: file didn't exist in 2016

Comment (NakulManchanda) [reply WebContent/js/utilities/fileHelper.js]: made it 2020

File: WebContent/js/components/dialogs/datasets/RenameDialog.js (hunk 8 above)
Comment (NakulManchanda) [reply WebContent/js/components/dialogs/datasets/RenameDialog.js]: made it 2020

File: WebContent/js/components/dialogs/datasets/RenameDatasetDialog.js (hunk 7 above)
Comment (NakulManchanda) [reply WebContent/js/components/dialogs/datasets/RenameDatasetDialog.js]: made it 2020

File: WebContent/js/components/dialogs/datasets/RenameDialog.js (hunk 6 above)
Comment (NakulManchanda) [reply WebContent/js/components/dialogs/datasets/RenameDialog.js]: added fullwidth prop

**Pull request merged**